### PR TITLE
[PNI] Articles nav link should be highlighted

### PR DESCRIPTION
**Pull request by robdivincenzo**
# Description

This pull request modifies the path check to see if the target is within the current path. It was previously only checking if the target was equal to the current path, and hence only worked on /articles pages.

Link to sample test page: https://foundation-s-11126-pni--c2dafh.herokuapp.com/en/privacynotincluded/
Related PRs/issues: #11126

# To Test

Pulled the acceptance criteria Mavis provided in the issue:

1) Navigate to the /articles category page. Make sure the "Articles" nav link is highlighted.
2) Visit an individual article. Make sure that the "Articles" nav link is highlighted.

┆Issue is synchronized with this [Jira Story](https://mozilla-hub.atlassian.net/browse/TP-297)

### CODE REVIEW

**Review by mmmavis — CHANGES_REQUESTED** [review]
I'm afraid we will need to come up with another solution. This fix is creating issues on About pages: https://foundation-s-11126-pni--c2dafh.herokuapp.com/en/privacynotincluded/about/why/ (multiple links on the side nav are highlighted)

File: network-api/networkapi/templates/fragments/buyersguide/about_nav.html [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
+    {% routablepageurl page 'about-why-view' as about_why_url %}
     {% routablepageurl page 'methodology-view' as methodology_url %}
     {% routablepageurl page 'contact-view' as contact_url %}
     {% routablepageurl page 'press-view' as press_url %}
 
     <div class="col-lg-3">
         <nav class="vertical-nav mt-5 pt-sm-2" title="{% trans "about sections" context "Tooltip on menu items" %}">
-            <div class="multipage-link-wrapper"><a class="multipage-link about-link {% bg_active_nav current_url about_why_url %}" href="{{ about_why_url }}">{% trans "Why we made this guide" %}</a></div>
             <div class="multipage-link-wrapper"><a class="multipage-link about-link {% bg_active_nav current_url how_to_use_url %}" href="{{ how_to_use_url }}">{% trans "How to use this guide" %}</a></div>
+            <div class="multipage-link-wrapper"><a class="multipage-link about-link {% bg_active_nav current_url about_why_url %}" href="{{ about_why_url }}">{% trans "Why we made this guide" %}</a></div>
```
Comment (mmmavis) [inline network-api/networkapi/templates/fragments/buyersguide/about_nav.html:13]: Hi @robdivincenzo , thanks for updating the PR. I just wanted to double check if stakeholders are okay with "Why we made this guide" being shifted down to the list.

Comment (robdivincenzo) [reply network-api/networkapi/templates/fragments/buyersguide/about_nav.html:13]: Good thought, let me double-check on that w/ @kristinashu as well

Comment (robdivincenzo) [reply network-api/networkapi/templates/fragments/buyersguide/about_nav.html:13]: Hey @mmmavis, Kristina and Jen are both ok with the change. Thanks!

Comment (mmmavis) [reply network-api/networkapi/templates/fragments/buyersguide/about_nav.html:13]: Lovely! Thank you for letting stakeholders aware of this change.

**Review by mmmavis — APPROVED** [review]
🎉

**Pull request merged**